Automattic/wp-calypso
# Launch nav-unification for all new users and 5% of existing users.

#### Changes proposed in this Pull Request

* Adds two util functions `isNewNavUnificationUser`, `isNavUnificationEnabledForUser`
* Hides the `nav-unification` modal for new users
* `LAUNCH_DATE` and `CURRENT_ROLLOUT_SEGMENT_PERCENTAGE` can be changed in `client/my-sites/sidebar-unified/utils.jsx`
* Enables `nav-unification` for all new users and 5% of existing users.

#### Testing instructions



By dispatching the following actions in your Redux dev tools you can check the desired behavior. Make sure you delete the nav-unification modal preference first 
![](https://cln.sh/Gbh3fc+)

**On your a8c account**
- you should have the feature enabled and the modal shown (if you haven'd dismissed it already)
- running the following, should dismiss the modal (mocks a11n registered after launch)
```
{
    type: 'CURRENT_USER_RECEIVE',
    user: {
            "ID": 202731081,
          }
}
```

**On an non a8c account**
- if your user id ends in 00,01,02,03,04 you should have the feature enabled and the modal shown
- Running the following should dismiss the modal
```
{
    type: 'CURRENT_USER_RECEIVE',
    user: {
            "ID": 202731081,
          }
}
```

- if your user id doesn't end in 00,01,02,03,04 you should have the feature disabled
- running the following, should enable nav-unification with the modal being shown (mocks a user in the rollout segment registered before launch)
```
{
    type: 'CURRENT_USER_RECEIVE',
    user: {
            "ID": 202731001,
          }
}
```

- running the following, should enable nav-unification **without** the modal being shown (mocks a user registered after launch)
```
{
    type: 'CURRENT_USER_RECEIVE',
    user: {
            "ID": 202731081,
            "date": "2021-03-11T18:54:26+00:00",
          }
}
```



Fixes #50379
Fixes #50037
